### PR TITLE
Sign executables on Mac.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: src/compiler/executable.cc [hunk 1]
```diff
@@ -44,6 +51,34 @@ static const uint8 VESSEL_TOKEN[] = { VESSEL_TOKEN_VALUES };
...
+  if (posix_spawnp(&child_pid, "codesign", null, null, argv, environ) != 0) goto fail;
+  do {
+    if (waitpid(child_pid, &status, 0) == -1) goto fail;
+  } while (WIFEXITED(status) == 0 && WIFSIGNALED(status) == 0);
+
+  free(out_path_mutable);
+  if (WIFEXITED(status) != 0) return WEXITSTATUS(status);
+  if (WIFSIGNALED(status) != 0) return -1;
+
+  fail:
```
Comment (kasperl) [inline src/compiler/executable.cc:75]: I'd indent the label by only one space and let the following lines have only two space indent.

File: src/compiler/executable.cc [hunk 2]
```diff
@@ -44,6 +51,34 @@ static const uint8 VESSEL_TOKEN[] = { VESSEL_TOKEN_VALUES };
 // just much more complicated for something that doesn't change that frequently.
 static int VESSEL_SIZES[] = { 128, 256, 512, 1024, 8192, };
 
+static int sign_if_necessary(const char* out_path) {
+#ifndef TOIT_DARWIN
+  return 0;
+#else
+  char codesign[] = { "codesign" };
```
Comment (kasperl) [inline src/compiler/executable.cc:58]: How does one install codesign? Do we need to update a README?

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: src/compiler/executable.cc (hunk 2 above)
Comment (floitsch) [reply src/compiler/executable.cc:58]: I think/hope it's preinstalled. If it isn't I would probably just ship the zsign executable with Toit instead.
This is the approach I would take for cross-compilation too. (That said: currently we ship with non-ARM executables, so the precompiled executables shouldn't complain).

File: src/compiler/executable.cc (hunk 1 above)
Comment (floitsch) [reply src/compiler/executable.cc]: done.

**Pull request merged**